### PR TITLE
Add Qwen2.5 to live docs

### PR DESCRIPTION
**Pull request by RdoubleA**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [x] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
*

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

File: docs/source/api_ref_models.rst [hunk 1]
```diff
@@ -211,7 +211,8 @@ To download the CodeLlama-7B model:
 qwen-2
 ------
 
-Models of size 0.5B, 1.5B, and 7B from the `Qwen2 family <https://huggingface.co/collections/Qwen/qwen2-6659360b33528ced941e557f>`_.
+Models of size 0.5B, 1.5B, 3B, 7B, 14B, 32B, 72B from the `Qwen2.5 family <https://huggingface.co/collections/Qwen/qwen25-66e81a666513e518adb90d9e>`
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_models.rst]:
```suggestion
Models of size 0.5B, 1.5B, 3B, 7B, 14B, 32B, 72B from the `Qwen2.5 family <https://huggingface.co/collections/Qwen/qwen25-66e81a666513e518adb90d9e>`_
```

Comment (RdoubleA) [reply docs/source/api_ref_models.rst]: I just beat you by 1 second >:)

File: docs/source/api_ref_models.rst [hunk 2]
```diff
@@ -225,14 +226,42 @@ To download the Qwen2 1.5B model, for example:
...
+    qwen2_5.lora_qwen2_5_14b_instruct
+    qwen2_5.qwen2_5_32b_base
+    qwen2_5.qwen2_5_32b_instruct
+    qwen2_5.lora_qwen2_5_32b_base
+    qwen2_5.lora_qwen2_5_32b_instruct
+    qwen2_5.qwen2_5_72b_base
+    qwen2_5.qwen2_5_72b_instruct
+    qwen2_5.lora_qwen2_5_72b_base
+    qwen2_5.lora_qwen2_5_72b_instruct
+    qwen2_5.qwen2_5_tokenizer
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_models.rst:263]: This is a bit hard on the eyes for me, I'd have to ctrl+f to navigate this.

Comment (SalmanMohammadi) [reply docs/source/api_ref_models.rst:263]: Is there also any reason not to split into separate Qwen 2, and Qwen 2.5 sections?

Comment (RdoubleA) [reply docs/source/api_ref_models.rst:263]: llama3 and 3.1 are in one section, so I was following that. but yeah I think the feature gap is significant enough for two separate sections

Comment (RdoubleA) [reply docs/source/api_ref_models.rst:263]: > This is a bit hard on the eyes for me, I'd have to ctrl+f to navigate this

not sure how else I can format it without just removing some builders

Comment (ebsmothers) [reply docs/source/api_ref_models.rst:263]: I think I missed when we put 3 and 3.1 together but not exactly sure why we did that tbh. Imo it is better to keep things separate, especially given the large number of Qwen2.5 models. As for readability, we should probably ideally have more nested subsections for different model sizes? I think we kinda shot ourselves in the foot by using section header for our title.. we should refactor to make it a proper title in RST parlance (stolen from stackoverflow):

```
==============
Document title
==============
-----------------
Document subtitle
-----------------

Section
=======

Sub-section
-----------
```

**Review by ebsmothers — APPROVED** [review]
Personally I would probably separate out Qwen2 and Qwen2.5 builders but given the Llama 3.1 precedent I don't have a strong opinion either way. Stamping to unblock

**Pull request merged**